### PR TITLE
Don't throw NotImplementedException #209

### PR DESCRIPTION
**Pull request by christian-sheridan**

### CODE REVIEW

File: Gu.Analyzers/Analyzers/ExceptionAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    internal class ExceptionAnalyzer : DiagnosticAnalyzer
+    {
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(GU0090DontThrowNotImplementedException.Descriptor);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.EnableConcurrentExecution();
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.ThrowExpression);
```
Comment (JohanLarsson) [inline Gu.Analyzers/Analyzers/ExceptionAnalyzer.cs]: We probably need to check other node types. IIRC `ThrowExpression` is for things like `?? throw ...`
The tests will reveal it, nice analyzer idea and the code looks nice.

Comment (christian-sheridan) [reply Gu.Analyzers/Analyzers/ExceptionAnalyzer.cs]: Yeah, I've switched it over to ``ThrowStatement``, Thanks.

**Comment by christian-sheridan** [comment]
Message: Gu.Roslyn.Asserts.AssertException : Mismatch at end.
Expected: | [GU0090](https://github.com/GuOrg/Gu.Analyzers/blob/master/documentation/GU0090.md)| Don't throw NotImplementedException.
Actual:   | [GU0083](https://github.com/GuOrg/Gu.Analyzers/blob/master/documentation/GU0083.md)| TestCase Arguments Mismatch Method Parameters

This is the test that is failing. I'm pretty sure this will be fixed when merging.

**Comment by JohanLarsson** [comment]
Nice one!

**Pull request merged**